### PR TITLE
[Build] Added flag to make go/pkg/ folder writable (removable)

### PR DESCRIPTION
**Pull request by vivekreddynv**
Fixes: https://github.com/Azure/sonic-buildimage/issues/8611

Signed-off-by: Vivek Reddy Karri <vkarri@nvidia.com>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -5,6 +5,7 @@ export GOROOT=/usr/local/go
 export GOPATH=$(HOME)/go
 export GOBIN=$(GOPATH)/bin
 export GO111MODULE=on
+export GOFLAGS=-modcacherw
```
Comment (prsunny) [inline Makefile:8]: does this compile for 201911/202012 branch?

Comment (vivekreddynv) [reply Makefile:8]: This option is available starting go version `1.14.x`.

I believe master & 202012 uses 1.14.x go version. 

Not sure about 201911. 

@sumukhatv, can you provide more  info about the versions the other images use?

Comment (sumukhatv) [reply Makefile:8]: we use go `1.14` in 201911: https://github.com/Azure/sonic-restapi/blob/8d09602a9e3c762f86992d8eaf642e7115c5318b/go-server-server/go.mod#L3

**Review by sumukhatv — APPROVED** [review]

**Comment by dgsudharsan** [comment]
@prsunny @sumukhatv  Can we merge this if we are good?

**Comment by liat-grozovik** [comment]
@prsunny could you please help to merge?
@dgsudharsan is this required for 202106 or master only?

**Comment by vivekreddynv** [comment]
> @prsunny could you please help to merge?
> @dgsudharsan is this required for 202106 or master only?

The change which caused the build failure is only in master and not in 202106

**Pull request merged**